dronekit_sitl/__init__.py

```diff
@@ -268,7 +347,23 @@ def cleanup_sim():
         # else:
         #     sitl = Popen(sitl_args, stdout=PIPE, stderr=PIPE)
 
-        p = Popen([self.path] + args, cwd=wd, shell=sys.platform == 'win32', stdout=PIPE, stderr=PIPE)
+        popen_args = []
+        if self.gdb:
+            commands_file = "/tmp/gdb.tmp"
```

Comment:
Writing that file out *might* work.  However, --gdb and --valgrind aren't going to work on Windows at all ATM; we pop up an xterm with gdb/valgrind running in that window.  I don't know if it is possible to get the same effect with "start" under Windows.